app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,25 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.find_by(token: params[:id])
+
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path
+    end
+  end
+
+  def create
+    user = User.find_by(token: params[:token])
+
+    if user
+      user.password = params[:password]
+      user.generate_token
+      user.save
```

Comment:
when you set `user.password =` and then `user.save` it is the same as `user.update(password: ...)`.

Another problem with this workflow is, what happens if the password is invalid? What do you return to user? Our solution in this isn't very bullet-proof, try to think about the fix.